### PR TITLE
Infinite scroll THE SAGA

### PR DESCRIPTION
**Pull request by DarthHater**
This adds infinite scroll to the ThreadList component, and a bit of stuff to track state via Redux on entry and exit into the ThreadList.

### CODE REVIEW

File: src/components/Thread/Thread.js [hunk 1]
```diff
@@ -23,9 +24,25 @@ class Thread extends Component {
         this.props.dispatch(threadActions.loadThread(id));
     }
 
+    componentDidMount = () => {
```
Comment (conwaydev) [inline src/components/Thread/Thread.js]: dont need this

File: src/components/Thread/Thread.js [hunk 2]
```diff
@@ -23,9 +24,25 @@ class Thread extends Component {
...
+    componentWillUnmount() {
+        this.props.dispatch(threadActions.exitThreadView());
+        this.props.dispatch(threadActions.exitPostsView());
+    }
+
     render() {
         return (
-            <div className='container'>
+            <div
+                style= {{
```
Comment (conwaydev) [inline src/components/Thread/Thread.js:35]: eh we should still use css for this stuff, scott or i can go through and clean though

File: src/components/ThreadList/ThreadList.js [hunk 3]
```diff
@@ -12,6 +12,7 @@ import { threadActions } from '../../actions';
 import * as permissions from '../../constants/permissions';
 import Grid from '@material-ui/core/Grid';
 import Button from '@material-ui/core/Button';
+import * as debounce from '../../helpers/debounce';
```
Comment (conwaydev) [inline src/components/ThreadList/ThreadList.js]: `import {debounce} from '../../helpers/debounce'`

File: src/components/ThreadList/ThreadList.js [hunk 4]
```diff
@@ -20,72 +21,135 @@ class ThreadList extends Component {
...
+            canDeleteThreads: auth.userHasPermission(permissions.deleteThread),
+            loading: false,
+            scrollPosition: 0
         }
     }
 
     componentDidMount() {
-        this.props.dispatch(threadActions.loadThreads());
+        this.props.dispatch(threadActions.enterThreadList(false));
+        window.addEventListener('scroll', debounce.debounce(this.handleScroll, 250));
```
Comment (conwaydev) [inline src/components/ThreadList/ThreadList.js]: if we want to use the scroll watcher lets put a passive flag on it: https://developers.google.com/web/updates/2016/06/passive-event-listeners

**Review by conwaydev — APPROVED** [review]

File: src/components/Thread/Thread.js (hunk 2 above)
Comment (DarthHater) [reply src/components/Thread/Thread.js:35]: Yeah we need to either figure out if we are going to use scss, css modules, or something akin. Go for it :)

**Pull request merged**